lib/plan-builder-base.js

```diff
@@ -122,11 +122,11 @@ function castArg(arg, funcName, paramName, argPos, paramTypes) {
         const value = arg[key];
         switch(key) {
           case 'scoreMethod':
-            if (['logtfidf', 'logtf', 'simple', 'score-logtfidf', 'score-logtf', 'score-simple'].includes(value)) {
+            if (['logtfidf', 'logtf', 'simple', 'score-logtfidf', 'score-logtf', 'score-simple', 'zero', 'random', 'bm25'].includes(value)) {
               return true;
             }
             throw new Error(
-                `${argLabel(funcName, paramName, argPos)} can only be logtfidf, logtf, or simple`
+                `${argLabel(funcName, paramName, argPos)} can only be 'logtfidf', 'logtf', 'simple', 'zero', 'random' or 'bm25'`
```

Comment:
Sorry, I should have noticed this initially - the values that are checked above and the values listed here are not the same. 

Based on what's in plan-search.js, looks like `simple` is valid but not `score-simple`. I'm assuming that the 3 values checked on line 125 that start with `score-` are not valid and should be removed. That will result in both lists having a length of 6. That also matches what's at https://docs.marklogic.com/cts:search , except that while the low-level function requires "score-" as a prefix, the REST API does not.